src/main/java/uk/gov/ons/ctp/response/action/service/ProcessEventService.java

```diff
@@ -135,6 +141,44 @@ public void processEvents(UUID collectionExerciseId, String eventTag) {
           .with("event", eventTag)
           .info("No Letters to process");
     }
+    partialProcessCheckPoint(collectionExerciseId, eventTag, partialEntry, instant);
+  }
+
+  private void partialProcessCheckPoint(
+      UUID collectionExerciseId,
+      String eventTag,
+      Optional<ActionEventPartialEntry> partialEntry,
+      Instant instant) {
+    Long numberOfActionCases =
+        actionCaseRepository.findByCollectionExerciseId(collectionExerciseId);
+    Long numberOfCases = caseSvcClientService.getNumberOfCases(collectionExerciseId);
```

Comment:
The getNumberOfCases method can return Null if it errors, but you're not checking for that here so you risk a null pointer if case service is down.

Reply:
👍